tree: changes needed to support Ansible 2.4

miabbott opened the pull request:
The main intention of these changes is to get the tests to a point where they can be used with Ansible 2.4 without any errors/warnings.  In order to get there, I had to update some of the roles along the way.

Specifically, I updated the `atomic` install/uninstall related roles to use unique variables names to prevent collisions during tests.  I also updated the `rpm_version` role to be more specific where tasks were being included from and how variables were being passed to `include_tasks`.

With those changes in place, I was able to update the tests to use `include_tasks` where a bare `include` was used.

miabbott commented:
This is a follow-on to #372, which needs to be reviewed and merged first.  I'll rebase after that PR gets in and this can be reviewed then.

As such, this also addresses #336

miabbott commented:
Ah, this will be fun...chicken and egg problem where `include_tasks` isn't supported until Ansible 2.4.  But the PAPR CI uses Ansible 2.2.

I guess I could just update `requirements.txt` to use 2.4 as part of this PR...

mike-nguyen commented:
I meant to leave a comment on #372 about `include_task`.  I'm not sure how merging these separate PRs might impact the tests with backwards compatibility.

mike-nguyen commented on roles/ostree_admin_unlock/tasks/main.yml:
```diff
@@ -7,7 +7,8 @@
   command: ostree admin unlock
 
-- include: '../../rpm_ostree_status/tasks/main.yml'
+- include_role:
```
Let's use import_role/import_task as much as possible and only use include_role/include_task when necessary.

mike-nguyen commented on roles/ostree_admin_unlock_hotfix/tasks/main.yml:
```diff
@@ -6,7 +6,8 @@
   command: ostree admin unlock --hotfix
 
-- include: '../../rpm_ostree_status/tasks/main.yml'
+- include_role:
```
same here

mike-nguyen commented on roles/rpm_version/tasks/main.yml:
```diff
@@ -7,6 +7,7 @@
 #
-- include: version.yml
-  rv_rpm: "{{ item }}"
+- include_tasks: roles/rpm_version/tasks/version.yml
```
same here

mike-nguyen commented on tests/docker-build-httpd/main.yml:
```diff
@@ -139,22 +139,22 @@
     # I would have used a 'block' of 'roles' here, but that is not
     # compatible with the 'with_items' loop control, so I stuck all
-    # the tasks into a single role file and used an 'include:'
+    # the tasks into a single role file and used an 'include_tasks:'
```
import_tasks

mike-nguyen commented on tests/docker-build-httpd/main.yml:
```diff
@@ -139,22 +139,22 @@
     # ansible-playbook -e "cli_skipped_images=busybox_httpd,debian_httpd"
     #
-    - include: tasks/build_run_remove.yml
+    - include_tasks: tasks/build_run_remove.yml
```
same

mike-nguyen replied:
this is a loop too so include_tasks is fine

mike-nguyen commented on tests/docker-build-httpd/main.yml:
```diff
@@ -139,22 +139,22 @@
 
     - when: ansible_distribution == "RedHat"
-      include: tasks/build_run_remove.yml
+      include_tasks: tasks/build_run_remove.yml
```
getting redundant lol.  do it everywhere!

miabbott commented:
CI is a mess because of broken versions of 'atomic' and unsupported tests on F27.  I think the playbooks themselves should be sound; they at least pass a syntax check.

I'll be running all the changed playbooks against some hosts to verify nothing has broken.

miabbott commented:
I've run all the supported tests against a F27 host without errors.  

I'll continue to run against other platforms, but I think it's in a place where it could be fully reviewed/tested by others.

mike-nguyen commented:
@miabbott can you resolve the tests/rpm-ostree/main.yml conflict?  I think it will re-run the CI #379 merged

miabbott commented:
@mike-nguyen I rebased on to `upstream/master`, squashed the fixups, and force pushed.  I think for this set of commits, we should do a 'rebase and merge' rather than 'squash and merge' - it should make tracking the changes a bit easier, IMO.

mike-nguyen commented:
@miabbott Rebase and merge sounds good based on the amount of changes in this PR.  I'll take a look when the CI finishes up.

mike-nguyen commented on tests/system-containers/main.yml:
```diff
@@ -127,9 +124,10 @@
       command: systemctl start {{ g_hw_name }}
 
-    - include: 'roles/atomic_system_install_verify/tasks/main.yml'
+    - include_role:
```
Is there a reason why we can't use import_role here? and below?

mike-nguyen commented on tests/system-containers/main.yml:
```diff
@@ -194,9 +194,10 @@
       command: systemctl start {{ g_hw_name }}
 
-    - include: 'roles/atomic_system_install_verify/tasks/main.yml'
+    - include_role:
```
import_role here

mike-nguyen commented on tests/system-containers/main.yml:
```diff
@@ -243,9 +246,10 @@
       command: systemctl start {{ g_hw_name }}
 
-    - include: 'roles/atomic_system_install_verify/tasks/main.yml'
+    - include_role:
```
import_role

mike-nguyen commented on tests/system-containers/main.yml:
```diff
@@ -286,13 +290,15 @@
           Actual: {{ c_output.stdout }}
 
-    - include: 'roles/atomic_system_uninstall/tasks/main.yml'
+    - include_role:
```
import_role

mike-nguyen commented on tests/system-containers/main.yml:
```diff
@@ -286,13 +290,15 @@
+        asu_name: "{{ g_hw_name }}"
 
-    - include: 'roles/atomic_system_uninstall_verify/tasks/main.yml'
+    - include_role:
```
import_role

mike-nguyen commented on tests/system-containers/main.yml:
```diff
@@ -320,9 +326,10 @@
       command: systemctl start {{ g_hw_name }}
 
-    - include: 'roles/atomic_system_install_verify/tasks/main.yml'
+    - include_role:
```
import_role

miabbott commented on roles/rpm_version/tasks/main.yml:
```diff
@@ -7,6 +7,6 @@
 # dict named g_atomic_host
 #
-- include: version.yml
+- include_tasks: version.yml
```
Are you sure we can safely use that here?  I thought since we are looping through the items, we have to use `include_tasks` to be able to get the correct value of `item` each time.

mike-nguyen replied:
I think you're right with include in loops.

miabbott commented:
Squashed a fixup into the existing commits and force pushed them all again.  :arrow_up:

mike-nguyen commented:
bot, retest this please

mike-nguyen commented:
bot, retest this please

mike-nguyen commented:
The failures aren't all consistent.  I'm going to look at this locally to see what's up.

miabbott commented:
@mike-nguyen Sorry it took me so long to report back (doing a lot of context switching today).

I ran the tests in order on the same CentOS AH 7 VM like the PAPR does and encountered similar failures, but they aren't related to the the changes themselves.  I think the some of the tests are leaving a "dirty" state on the VM and it is tripping up the subsequent tests.

Having said all that, I'm comfortable with merging this if you are.

mike-nguyen commented:
@miabbott I ran  some tests on centos last night to see if I could get a clean run but ran into a lot of failures.  I'm going to take a look and if they aren't related to these changes, I'm going to merge.
